### PR TITLE
[master] Semaphore Auto Pin Update

### PR DESCRIPTION
**Pull request by marvin-tigera**

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

File: tests/fv/fv_test.go
Comment (caseydavenport) [inline tests/fv/fv_test.go:308]: This step was unnecessary since the AssignIP call does it automatically below.

It also left the datastore in an intermediate state such that the new IPAM controller changes detected as a candidate for garbage collection.

**Review by mgleung — APPROVED** [review]
LGTM

**Pull request merged**
